ValveAnnulusAnalysis/HeartValveWidgets/ValveSequenceBrowserWidget.py

```diff
@@ -0,0 +1,267 @@
+
+  def _removeHeartValveBrowserNodeObserver(self):
+    if self.valveBrowserNode and self.valveBrowserNodeObserver:
+      self.valveBrowserNode.RemoveObserver(self.valveBrowserNodeObserver)
+      self.valveBrowserNodeObserver = None
+
+  def _setAndObserveHeartValveNode(self, heartValveNode):
+    self._removeHeartValveNodeObserver()
+    self.heartValveNode = heartValveNode
+    if self.heartValveNode:
+      self.heartValveNodeObserver = \
+        self.heartValveNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.updateGUIFromHeartValveNode)
+
+  def _removeHeartValveNodeObserver(self):
+    if self.heartValveNode and self.heartValveNodeObserver:
+      self.heartValveNode.RemoveObserver(self.heartValveNodeObserver)
+      self.heartValveNodeObserver = None
+
+  def onGoToAnalyzedFrameButtonClicked(self):
+    HeartValveLib.goToAnalyzedFrame(self.valveBrowser.valveModel)
+
+  def _addValveVolumeBrowserNodeObserver(self, valveVolumeBrowserNode):
+    self._removeValveVolumeBrowserObserver()
+    # Set and observe new parameter node
+    self.valveVolumeBrowserNode = valveVolumeBrowserNode
+    if self.valveVolumeBrowserNode:
+      self.valveVolumeBrowserNodeObserver = \
+        self.valveVolumeBrowserNode.AddObserver(vtk.vtkCommand.ModifiedEvent, self.updateGUIFromValveVolumeBrowser)
+
+  def _removeValveVolumeBrowserObserver(self):
+    if self.valveVolumeBrowserNode and self.valveVolumeBrowserNodeObserver:
+      self.valveVolumeBrowserNode.RemoveObserver(self.valveVolumeBrowserNodeObserver)
+      self.valveVolumeBrowserNodeObserver = None
+
+  def onCardiacCyclePhaseChanged(self):
+    if self.valveModel is None:
+      return
+    self.valveModel.setCardiacCyclePhase(self.ui.cardiacCyclePhaseSelector.currentText)
+
+  def onAddTimePointButtonClicked(self):
+    volumeSequenceIndex, volumeSequenceIndexValue = self.valveBrowser.getDisplayedValveVolumeSequenceIndexAndValue()
+    heartValveSequenceIndex = self.valveBrowser.heartValveSequenceNode.GetItemNumberFromIndexValue(volumeSequenceIndexValue)
+
+    if heartValveSequenceIndex >= 0: # item for timepoint exists
+      self.valveBrowser.valveBrowserNode.SetSelectedItemNumber(heartValveSequenceIndex)
+    else:
+      logging.info(f"Add time point")
+      if not volumeSequenceIndexValue:
+        raise RuntimeError("Failed to add time point, could not get volume sequence")
+      self.valveBrowser.addTimePoint(volumeSequenceIndexValue)
+      heartValveNode = self.valveBrowser.heartValveNode if self.valveBrowser else None
+      self.setHeartValveNode(heartValveNode)
+
+    self.updateGUIFromValveVolumeBrowser()
+
+  def onRemoveTimePointButtonClicked(self):
+    itemIndex, indexValue = self.valveBrowser.getDisplayedHeartValveSequenceIndexAndValue()
+    if indexValue is not None:
+      # TODO: add confirm dialog
+      self.valveBrowser.removeTimePoint(indexValue)
+
+    self.updateGUIFromHeartValveNode()
+
+  def disableGUI(self):
```

Comment:
If there is a `disableGUI` method then there must be an `enableGUI` counterpart but then it is simpler to have a single, positive `enableGUI(enable)` method.